bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/handler/RadioThermostatHandler.java

```diff
@@ -0,0 +1,451 @@
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(RadioThermostatThingActions.class);
+    }
+
+    /**
+     * Start the job to periodically update data from the thermostat
+     */
+    private void startAutomaticRefresh() {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                // send an async call to the thermostat to get the 'tstat' data
+                connector.getAsyncThermostatData(DEFAULT_RESOURCE);
+            };
+
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Start the job to periodically update humidity and runtime date from the thermostat
+     */
+    private void startAutomaticLogRefresh() {
+        if (logRefreshJob == null || logRefreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                // Request humidity data from the thermostat if we are a CT80
+                if (config.isCT80) {
+                    // send an async call to the thermostat to get the humidity data
+                    connector.getAsyncThermostatData(HUMIDITY_RESOURCE);
+                }
+
+                if (!config.disableLogs) {
+                    // send an async call to the thermostat to get the runtime data
+                    connector.getAsyncThermostatData(RUNTIME_RESOURCE);
+                }
+            };
+
+            int delay = ((config.logRefresh != null) ? config.logRefresh.intValue() : DEFAULT_LOG_REFRESH_PERIOD) * 60;
+            logRefreshJob = scheduler.scheduleWithFixedDelay(runnable, 30, delay, TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the RadioThermostat handler.");
+        connector.removeEventListener(this);
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+        if (logRefreshJob != null && !logRefreshJob.isCancelled()) {
+            logRefreshJob.cancel(true);
+            logRefreshJob = null;
+        }
+    }
+
+    public void handleRawCommand(@Nullable String rawCommand) {
+        connector.sendCommand(null, null, rawCommand);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            updateChannel(channelUID.getId(), rthermData);
+        } else {
+            Integer cmdInt = -1;
+            String cmdStr = command.toString();
+            if (cmdStr != null) {
+                try {
+                    // remove all non-numeric characters except negative '-' and parse int
+                    cmdInt = Integer.parseInt(cmdStr.replaceAll("[^\\d-]", ""));
```

Comment:
Things might get wonky if the user supplies a value with a decimal point.

Reply:
They shouldn't. The values are all constrained by the drop downs and set points.

Reply:
What if the command comes from someone's rules?

Reply:
Alright, I improved it. A number with a decimal will be transformed into an appropriate integer value. If their rules are sending values outside the range of what the thermostat will accept, its on them.